src/brsTypes/components/RoSGNode.ts

```diff
@@ -268,11 +268,34 @@ export class Field {
     }
 }
 
+/* Hierarchy of all node Types. Used to discover is a current node is a subtype of another node */
+const subtypeHierarchy = new Map<string, string>();
+
+/**
+ *  Checks the node sub type hierarchy to see if the current node is a sub component of the given node type
+ *
+ * @param {string} currentNodeType
+ * @param {string} checkType
+ * @returns {boolean}
+ */
+function isSubtypeCheck(currentNodeType: string, checkType: string): boolean {
+    if (!currentNodeType || !checkType) {
+        return false;
+    }
+    checkType = checkType.toLowerCase();
+    currentNodeType = currentNodeType.toLowerCase();
+    if (currentNodeType === checkType) {
+        return true;
+    }
+    return isSubtypeCheck(subtypeHierarchy.get(currentNodeType)!, checkType);
+}
```

Comment:
The coercion to a non-null type with `!` feels strange in combination with the checks in line 282 and the `: string` typing for the `currentNodeType` and `checkType` parameters.  We force the type system to think the next value of `currentNodeType` is truthy, then in that next call to `isSubtypeCheck` we check for falsy values immediately. Consider checking checking the result of `subtypeHierarchy.get()` before `isSubtypeCheck` calls itself recursively:

```suggestion
function isSubtypeCheck(currentNodeType: string, checkType: string): boolean {
    checkType = checkType.toLowerCase();
    currentNodeType = currentNodeType.toLowerCase();
    if (currentNodeType === checkType) {
        return true;
    }
    let nextNodeType = subtypeHierarchy.get(currentNodeType);
    if (nextNodeType == null) {
        return false;
    }
    return isSubtypeCheck(nextNodeType, checkType);
}
```